### DISCUSSION TITLE
How to dynamically set default values

### DISCUSSION THREAD

**Discussion by rburgst** [keystonejs/keystone#7254] [Questions] [answered]
I would like to be able to "calculate" a default value (specifically for relationships). I think this would be a useful feature.

Right now I would need to implement my own custom field to achieve something like this.

**Comment by dcousens** [comment]
~@rburgst do the examples in https://github.com/keystonejs/keystone/tree/main/examples/default-values#dynamic help you with this?~

**Reply by rburgst** [reply]
It does. I feel like a doofus. However to my defense, this page https://keystonejs.com/docs/apis/fields#relationship-type does not make any mention of it.

**Reply by dcousens** [reply]
Thanks @rburgst - we'll look to try and improve that documentation!  Thank you for opening this issue.

**Reply by rburgst** [reply]
Actually the implementation and the documentation do not match in https://github.com/keystonejs/keystone/tree/main/examples/default-values#dynamic

The documentation mentions

```

We use context along with the Query API to set the default assignee of a task to be a user named "Anonymous".

      assignedTo: relationship({
        ref: 'Person.tasks',
        many: false,
        // Dynamic default: Find an anonymous user and assign the task to them
        defaultValue: async ({ context }) => {
          const anonymous = await context.query.Person.findMany({
            where: { name: 'Anonymous' },
          });
          if (anonymous.length > 0) {
            return { connect: { id: anonymous[0].id } };
          }
          // If we don't have an anonymous user return undefined so as not to apply any default
        },
      }),
```

however there is no `defaultValue` on the relationship. The actual implementation
https://github.com/keystonejs/keystone/blob/main/examples/default-values/schema.ts#L31-L49
uses `resolveInput` to achieve this.

This is quite a different user experience than I was looking for. I would like the dropdown fields in the relationship to be already pre-populated (just like the documentation suggests). The actual solution will leave the field empty and only fill it when you encounter the empty field upon creation.

**Reply by dcousens** [reply]
Sorry @rburgst,  this was previously a feature that had been removed in favour of `resolveInput` as the example code rightly demonstrates.
We will update that documentation as soon as possible.

See https://github.com/keystonejs/keystone/pull/7284 for removal of the deprecated content.

**Comment by raveling** [comment, marked as answer]
https://github.com/keystonejs/keystone/discussions/7254#discussioncomment-2177911
